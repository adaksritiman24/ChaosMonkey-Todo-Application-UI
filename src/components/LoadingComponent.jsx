import { Grid, Skeleton, Stack } from '@mui/material'
import { grey } from '@mui/material/colors'
import { Box } from '@mui/system'
import React from 'react'

const LoadingItem = ()=>
    (
        <Grid item md={4} sm={6} xs={12}>
            <Box border={1} borderColor={grey[100]}
                sx={{
                    height : "250px"
                }}
            >   
            <Stack spacing={2} m={1} mt={2}>
                <Skeleton variant='rectangular' height="40px" width="80%"  sx={{ borderRadius : "4px",}}/>
                <Skeleton variant='rectangular' height="95px"  sx={{ borderRadius : "4px",}}/>
                <Stack direction="row" spacing={1} sx={{paddingTop:"10px"}}>
                    <Skeleton variant='rectangular' width="90px"  height="35px" sx={{ borderRadius : "20px",}}/>
                    <Skeleton variant='rectangular' width="90px"  height="35px" sx={{ borderRadius : "20px",}}/>
                </Stack>
            </Stack>
            </Box>
        </Grid>
    )

const LoadingComponent=()=> {

  return (
      <>
    <LoadingItem/>
    <LoadingItem/>
    <LoadingItem/>
    <LoadingItem/>
    <LoadingItem/>
    <LoadingItem/>
    </>
  )
}

export default LoadingComponent