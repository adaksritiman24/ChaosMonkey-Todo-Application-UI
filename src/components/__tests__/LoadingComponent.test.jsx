import { render, screen } from "@testing-library/react"
import LoadingComponent from "../LoadingComponent"


describe("Loading Component", ()=> {

    it("Should show nine Loading items", ()=>{
        render(<LoadingComponent/>);

        expect(screen.getAllByTestId("loading-box")).toHaveLength(9);
    })

    it("should match snapshots",()=>{
        const {baseElement} = render(<LoadingComponent/>);

        expect(baseElement).toMatchSnapshot();
    })
})