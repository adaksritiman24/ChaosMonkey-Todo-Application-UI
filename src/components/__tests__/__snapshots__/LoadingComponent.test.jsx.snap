// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should match snapshots 1`] = `
<body>
  <div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-lg-4 css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1tesdz0"
        data-testid="loading-box"
      >
        <div
          class="css-1nva18r-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="width: 80%; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-tqb5fd-MuiSkeleton-root"
            style="height: 95px;"
          />
          <div
            class="css-1y2jz8o-MuiStack-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-flnd2e-MuiSkeleton-root"
              style="width: 90px; height: 35px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
